dev/Repeater/APITests/RepeaterTests.cs

```diff
@@ -234,6 +234,78 @@ public void NestedRepeaterWithDataTemplateScenario()
             NestedRepeaterWithDataTemplateScenario(disableAnimation: false);
         }
 
+
+        [TestMethod]
+        public void FocusedItemGetsRecycledUponCollectionReset()
+        {
+            List<object> ResettingListItems = new List<object> { "item0", "item1", "item2", "item3", "item4", "item5", "item6", "item7", "item8", "item9" };
+            ItemsRepeater repeater;
+            RunOnUIThread.Execute(() => {
+
+                var template = (DataTemplate)XamlReader.Load(
+                        @"<DataTemplate xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'> 
+                            <Button AutomationProperties.Name='{Binding}' Content='{Binding}'/>
+                        </DataTemplate>");
+
+                repeater = new ItemsRepeater() {
+                    ItemsSource = ResettingListItems,
+                    ItemTemplate = template
+                };
+
+                Content = new ItemsRepeaterScrollHost() {
+                    Width = 400,
+                    Height = 800,
+                    ScrollViewer = new ScrollViewer {
+                        Content = repeater
+                    }
+                };
+
+                Content.UpdateLayout();
```

Comment:
close the RunOnUIThread here and do an IdleSync.wait outside the UI thread to intialize the repeater.